### ISSUE TITLE
@types imports don't need `/index` even with classic module resolution

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#23291]
**TypeScript Version:**  master

**Code**

With `"moduleResolution": "classic"`:

**node_modules/@types/foo/index.d.ts
```ts
export const xyz: number;
```

**a.ts**
```ts
xyz;
```

**Expected behavior:**

Import fix imports from "foo".

**Actual behavior:**

Import fix imports from "foo/index".

**Comment by mhegazy** [comment]
but the compiler will not resolve it correctly? in other words, the error will not go away..

**Comment by andy-ms** [comment]
No, we will resolve an import `foo` correctly if `node_modules/@types/foo/index.d.ts` exists, even with classic resolution.

**Comment by mhegazy** [comment]
ah, @types make sense.

**Comment by mhegazy** [comment]
but not other modules..
